pilot/lib/pilot/tester_manager.rb

```diff
@@ -91,21 +95,41 @@ def find_app(app_filter: nil)
       nil
     end
 
-    def find_or_create_tester(email: nil, first_name: nil, last_name: nil)
-      tester = Spaceship::Tunes::Tester::Internal.find(config[:email])
-      tester ||= Spaceship::Tunes::Tester::External.find(config[:email])
+    def find_or_create_tester(email: nil, first_name: nil, last_name: nil, app: nil)
```

Comment:
This method is quite long with lots of branching. I think it would be quite a pain to test. I think start writing some tests for it, it would uncover how we can decompose it into smaller pieces.